Content.Server/GameObjects/Components/GUI/HandsComponent.cs

```diff
@@ -176,16 +177,19 @@ public bool PutInHand(ItemComponent item, string index, bool fallback = true)
             return success;
         }
 
-        public void PutInHandOrDrop(ItemComponent item)
+        public void PutInHandOrDrop(ItemComponent item, bool mobCheck = true)
         {
-            if (!PutInHand(item))
+            if (!PutInHand(item, mobCheck))
             {
                 item.Owner.Transform.GridPosition = Owner.Transform.GridPosition;
             }
         }
 
-        public bool CanPutInHand(ItemComponent item)
+        public bool CanPutInHand(ItemComponent item, bool mobCheck = true)
         {
+            if (mobCheck && !ActionBlockerSystem.CanPickup(Owner))
+                return false;
+
             foreach (var handName in ActivePriorityEnumerable())
             {
                 if (CanPutInHand(item, handName))
```

Comment:
Does this need the mobCheck too?

Reply:
Not really. In fact, since we already do a mobCheck above, this was wasting cycles by doing mobChecks for each hand. Aside from completely ignoring `mobCheck`'s actual value. Fixed now!